### PR TITLE
[DX] PHPStan rule to check getNodeTypes() return classes of `PhpParser\Node`

### PR DESCRIPTION
**Pull request by dobryy**
Closes: #4105

### CODE REVIEW

**Comment by dobryy** [comment]
@TomasVotruba 
Seems like we have no such issues, that is why I've added the failing case https://github.com/rectorphp/rector/pull/4210/commits/e59dff8dc8af4c599c54ae5bf58ba5909b806afe.
Once you'll find that rule is okay, I'll fix it.

File: utils/phpstan-extensions/src/Rule/CheckGetNodeTypesReturnPhpParserNodeRule.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        $nodeFinder = new NodeFinder();
+        /** @var Return_[] $returns */
+        $returns = $nodeFinder->findInstanceOf($classMethod, Return_::class);
+
+        if (count($returns) === 1) {
+            return false;
+        }
+
+        throw new ShouldNotHappenException('More than one return statement exist');
```
Comment (dobryy) [inline utils/phpstan-extensions/src/Rule/CheckGetNodeTypesReturnPhpParserNodeRule.php:77]: @TomasVotruba, at the moment, we don't have such cases. Do you think I have to add handling of such a situation into this rule or some separate rule?

Comment (TomasVotruba) [reply utils/phpstan-extensions/src/Rule/CheckGetNodeTypesReturnPhpParserNodeRule.php:77]: I think exception here is the best solution :+1:

**Comment by TomasVotruba** [comment]
> Once you'll find that rule is okay, I'll fix it.

Very nice, thank you :+1:

**Review by TomasVotruba — APPROVED** [review]

**Comment by dobryy** [comment]
Can be merged.

**Pull request merged**